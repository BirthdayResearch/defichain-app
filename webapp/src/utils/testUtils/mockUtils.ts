import PersistentStore from '../persistentStore';
import * as isElectron from '../isElectron';
import * as Notifications from '../notifications';
import log from 'loglevel';
import { runSaga } from 'redux-saga';
import axios from 'axios';

const mockPersistentStore = (getData, setData) => {
  PersistentStore.get = jest.fn(() => getData);
  PersistentStore.set = jest.fn(() => setData);
  return PersistentStore;
};

const mockAxios = mockFunc => {
  axios.create = jest.fn().mockImplementation(() => ({
    post: mockFunc,
  }));
};
const dispatchedFunc = async (method, payload = {}) => {
  const dispatched = Array();
  let action = {};
  if (payload) action = { payload };
  await runSaga(
    {
      dispatch: action => dispatched.push(action),
    },
    method,
    action
  );
  return dispatched;
};

const mockisElectron = () => {
  return jest.spyOn(isElectron, 'isElectron');
};
const mockIpcRenderer = () => {
  return jest.spyOn(isElectron, 'ipcRendererFunc');
};

const spyLogger = () => {
  return jest.spyOn(log, 'error');
};

const mockNotification = () => {
  return (
    jest
      .spyOn(Notifications, 'default')
      // tslint:disable-next-line: no-empty
      .mockImplementation(() => {})
  );
};
export {
  mockPersistentStore,
  mockAxios,
  dispatchedFunc,
  mockisElectron,
  mockIpcRenderer,
  spyLogger,
  mockNotification,
};
