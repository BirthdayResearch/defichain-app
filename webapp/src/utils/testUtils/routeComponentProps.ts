import { createMemoryHistory, createLocation } from 'history';
import { match } from 'react-router';
const history = createMemoryHistory();
const path = `/route/:id`;
const match: match<{ id: string }> = {
  isExact: false,
  path,
  url: path.replace(':id', '1'),
  params: { id: '1' },
};

const location = createLocation(match.url);
delete history.location.key;
delete location.key;

export { history, match, location };
