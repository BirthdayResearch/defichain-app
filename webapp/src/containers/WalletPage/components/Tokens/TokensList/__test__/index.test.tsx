import React from 'react';
import WalletTokensList from '../index';
import { Provider } from 'react-redux';
import store from '../../../../../../app/rootStore';
import Enzyme from '../../../../../../utils/testUtils/enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import * as routeComponentProps from '../../../../../../utils/testUtils/routeComponentProps';

describe('WalletTokensList component', () => {
  it('should check for snapshot', () => {
    const wrapper = Enzyme.mount(
      <Router>
        <Provider store={store}>
          <WalletTokensList {...routeComponentProps} />
        </Provider>
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
