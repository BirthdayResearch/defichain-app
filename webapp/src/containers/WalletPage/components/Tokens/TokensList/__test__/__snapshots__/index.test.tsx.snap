// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletTokensList component should check for snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(WalletTokensList)
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/route/1",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": false,
            "params": Object {
              "id": "1",
            },
            "path": "/route/:id",
            "url": "/route/1",
          }
        }
      >
        <WalletTokensList
          accountTokens={Array []}
          fetchAccountTokensRequest={[Function]}
          fetchInstantBalanceRequest={[Function]}
          fetchTokensRequest={[Function]}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isAccountLoadingTokens={true}
          isAccountTokensLoaded={false}
          isLoadingTokens={true}
          isTokensLoaded={false}
          isWalletCreatedFlag={false}
          location={
            Object {
              "hash": "",
              "pathname": "/route/1",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": false,
              "params": Object {
                "id": "1",
              },
              "path": "/route/:id",
              "url": "/route/1",
            }
          }
          openResetWalletDatModal={false}
          tokens={Array []}
          unit="DFI"
          walletBalance={0}
        >
          <div>
            Loadingâ€¦
          </div>
        </WalletTokensList>
      </Connect(WalletTokensList)>
    </Provider>
  </Router>
</BrowserRouter>
`;
